presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -514,22 +514,75 @@ public synchronized MetastoreOperationResult renameTable(MetastoreContext metast
         requireNonNull(newTableName, "newTableName is null");
         Table table = getRequiredTable(metastoreContext, databaseName, tableName);
         getRequiredDatabase(metastoreContext, newDatabaseName);
-        if (isIcebergTable(table) && !isIcebergView(table)) {
-            throw new PrestoException(NOT_SUPPORTED, "Rename not supported for Iceberg tables");
-        }
         // verify new table does not exist
         verifyTableNotExists(metastoreContext, newDatabaseName, newTableName);
 
+        Path metadataDirectory = getTableMetadataDirectory(databaseName, tableName);
+        Path newMetadataDirectory = getTableMetadataDirectory(newDatabaseName, newTableName);
+
+        if (isIcebergTable(table)) {
+            renameIcebergTable(metadataDirectory, newMetadataDirectory);
+        }
+        else {
+            renameTable(metadataDirectory, newMetadataDirectory);
+        }
+
+        return EMPTY_RESULT;
+    }
+
+    private void renameIcebergTable(Path originalMetadataDirectory, Path newMetadataDirectory)
+    {
+        Optional<Runnable> rollbackAction = Optional.empty();
         try {
-            if (!metadataFileSystem.rename(getTableMetadataDirectory(databaseName, tableName), getTableMetadataDirectory(newDatabaseName, newTableName))) {
+            // If the directory `.prestoPermissions` exists, copy it to the new table metadata directory
+            if (metadataFileSystem.exists(new Path(originalMetadataDirectory, PRESTO_PERMISSIONS_DIRECTORY_NAME))) {
+                if (!FileUtil.copy(metadataFileSystem, new Path(originalMetadataDirectory, PRESTO_PERMISSIONS_DIRECTORY_NAME),
+                        metadataFileSystem, new Path(newMetadataDirectory, PRESTO_PERMISSIONS_DIRECTORY_NAME), false, metadataFileSystem.getConf())) {
+                    throw new PrestoException(HIVE_METASTORE_ERROR, "Could not rename table directory");
```

Comment:
Nit: We can add some more details in exception here around which table directory is being renamed